pkg/ccl/sqlccl/restore.go

```diff
@@ -1153,6 +1153,52 @@ func doRestorePlan(
 	return nil
 }
 
+func loadBackupSQLDescs(ctx context.Context, details jobs.RestoreDetails, settings *cluster.Settings) ([]BackupDescriptor, []sqlbase.Descriptor, error) {
+	backupDescs, err := loadBackupDescs(ctx, details.URIs, settings)
+	if err != nil {
+		return nil, nil, err
+	}
+	lastBackupDesc := backupDescs[len(backupDescs)-1]
+
+	var sqlDescs []sqlbase.Descriptor
+	for _, desc := range lastBackupDesc.Descriptors {
+		if _, ok := details.TableRewrites[desc.GetID()]; ok {
+			sqlDescs = append(sqlDescs, desc)
+		}
+	}
+	return backupDescs, sqlDescs, nil
+}
+
+// restoreFailHook removes KV data that has been committed from a restore that
+// has failed or been canceled. It does this by adding the table descriptors
+// in DROP state, which causes the schema change stuff to delete the keys
+// in the background.
+func restoreFailHook(ctx context.Context, txn *client.Txn, settings *cluster.Settings, details *jobs.RestoreDetails) error {
+	// Needed to trigger the schema change manager.
```

Comment:
More precisely it's needed to trigger a re-gossiping of the system config. Any transaction that touches the system config span is supposed to set this, which I'm realizing we don't do elsewhere in restore. I've filed #19617 to track, because `SetSystemConfigTrigger` confuses me every time I see it.